PyPortal_Tides/admiralty_tides.py

```diff
@@ -0,0 +1,105 @@
+import time
+import board
+from adafruit_pyportal import PyPortal
+from adafruit_bitmap_font import bitmap_font
+from adafruit_display_text.label import Label
+import json
+from secrets import secrets
+
+#--| USER CONFIG |--------------------------
+STATION_ID = "0245"   # tide location, find yours from admiralty website/
+HI_COLOR   = 0x00FF00    # high tide times color
+LO_COLOR   = 0x11FFFF    # low tide times color
+DATE_COLOR = 0xFFFFFF    # date and time color
+#-------------------------------------------
+
+# pylint: disable=line-too-long
+DATA_SOURCE = "https://admiraltyapi.azure-api.net/uktidalapi/api/V1/Stations/" + STATION_ID + "/TidalEvents?duration=1"
+DATA_LOCATION = []
+
+# determine the current working directory needed so we know where to find files
+cwd = ("/"+__file__).rsplit('/', 1)[0]
+pyportal = PyPortal(url=DATA_SOURCE,
+                    headers={"Ocp-Apim-Subscription-Key":secrets['Ocp-Apim-Subscription-Key']},
+                    json_path=DATA_LOCATION,
+                    status_neopixel=board.NEOPIXEL,
+                    default_bg=cwd+"/tides_bg.bmp")
+
+# Connect to the internet and get local time
+pyportal.get_local_time()
+
+# Setup tide times font
+tide_font = bitmap_font.load_font(cwd+"/fonts/cq-mono-30.bdf")
+tide_font.load_glyphs(b'1234567890:')
+
+# Setup date and time font
+date_font = bitmap_font.load_font(cwd+"/fonts/Arial-12.bdf")
+date_font.load_glyphs(b'1234567890-')
+
+# Labels setup
+HI_LABELS = [ Label(tide_font, text="00:00", color=HI_COLOR, x= 40, y= 80) ,
+              Label(tide_font, text="00:00", color=HI_COLOR, x= 40, y=165) ]
+LO_LABELS = [ Label(tide_font, text="00:00", color=LO_COLOR, x=180, y= 80) ,
+              Label(tide_font, text="00:00", color=LO_COLOR, x=180, y=165) ]
+DATE_LABEL = Label(date_font, text="0000-00-00 00:00:00", color=DATE_COLOR, x=75, y=228)
+
+# Add all the labels to the display
+for label in HI_LABELS + LO_LABELS + [DATE_LABEL]:
+    pyportal.splash.append(label)
+
+def get_tide_info():
+    """Fetch JSON tide time info and return it."""
+
+    # Get raw JSON data
+    raw_info = pyportal.fetch()
+    raw_info = json.loads(raw_info)
```

Comment:
Were you not able to work with the return from `pyportal.fetch()` directly for some reason?

Reply:
No it returned an array, that led to an error